### ISSUE TITLE
Update Crashlytics cause us a spike in Native Crash report (libc)

### ISSUE THREAD

**Issue by roscrazy** [firebase/firebase-android-sdk#4718]
Hello Firebase Crashlytics Team,

I have a question regarding the recent update we made in our android project, where we updated the firebase-crashlytics and firebase-crashlytics-ndk to version 18.3.2 from version 18.1.0. As there was a 1.5-year gap and 17 versions skipped, we noticed a spike of approximately 88% in the number of crashes and 100% in the number of impacted users in the **native (Libc) crashes**. We have also observed significant differences in the crashes distributed by the OS version before and after the update.

Before the update (Libc)
<img width="454" alt="before_spike" src="https://user-images.githubusercontent.com/6395178/221392691-04404c8b-77fd-46e5-bb54-b373add95e18.png">
After the SDK update (Libc)
<img width="448" alt="after_spike" src="https://user-images.githubusercontent.com/6395178/221392703-deea51ce-4c2b-4dcf-82a5-2a5c94478215.png">

We checked your release notes and found that you fixed [this issue](https://github.com/firebase/firebase-android-sdk/issues/4313) in version 18.3.2. However, the thread is canceled, and we cannot follow up. We have the following questions:

1. Does the Firebase crashlytic NDK 18.1.0 also get affected by [the issue](https://github.com/firebase/firebase-android-sdk/issues/4313)?
2. Would you have any update or fix for the 17 skipped versions from 18.1.0 to 18.3.2 that could have caused the spike?

I apologize if this is not an issue report, but we couldn't find any channel to ask these questions. Thank you in advance for your help.

**Comment by argzdev** [comment]
Thanks for reporting, @roscrazy. Could you share some stack trace of your issue? Also any chance you're able to reproduce the same behavior? If yes, could you share us a minimal repro or code snippet, so that we can investigate this further. Thanks!

**Comment by roscrazy** [comment]
Thank @argzdev for the reply. 
As shared, this is more likely a question, not an issue report. The native Libc crash stacktrace does not provide much info
```
Crashed: Thread: SIGABRT  0x0000000000000000
#00 pc 0x80fe0 libc.so 
#01 pc 0x80fb8 libc.so 
#02 pc 0x2a8a8 libui.so 
#03 pc 0x2a670 libui.so 
#04 pc 0xc8aac libgui.so 
#05 pc 0x7b34c libbinder.so 
#06 pc 0xe0d88 libgui.so 
#07 pc 0xe1c94 libgui.so 
#08 pc 0xe1df8 libgui.so 
#09 pc 0xe2c48 libgui.so 
#10 pc 0x54a4c libbinder.so 
#11 pc 0x5fcd4 libbinder.so 
#12 pc 0x5f810 libbinder.so 
#13 pc 0x600a4 libbinder.so 
#14 pc 0x90048 libbinder.so 
#15 pc 0x13550 libutils.so 
#16 pc 0xcc59c libandroid_runtime.so 
#17 pc 0xeb720 libc.so 
#18 pc 0x7e2d0 libc.so 
```

**Comment by argzdev** [comment]
Hi @roscrazy, sorry for the radio silence. Our Crashlytics team is dealing with a lot of issues as of the moment.

> Does the Firebase crashlytic NDK 18.1.0 also get affected by https://github.com/firebase/firebase-android-sdk/issues/4313?

No, the previous version should not be affected by that issue since the AGP was not yet updated for Crashlytics. Reading through the [filed report](https://github.com/firebase/firebase-android-sdk/pull/4311). The AGP have been updated most probably after the version of 18.1.0 and then the missing `libcrashlytics-trampoline.so` was added in version 18.3.2.

> Would you have any update or fix for the 17 skipped versions from 18.1.0 to 18.3.2 that could have caused the spike?

This is difficult to say, especially since the gap is too wide, it would be finding a needle in a haystack. A spike in crashes can be caused by multiple reasons and it might or might not be related to Crashlytics at all. If you'd like what we can do is, we can investigate the issue further with more details such as:
1. Steps to reproduce this issue
2. Minimal reproducible example or code snippets

Let me know if these help. Thanks!
